src/frontends/android/apiclient/src/main/java/org/strongswan/android/apiclient/ReturnMessenger.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © 2015 FancyFon Software Ltd.
+ * All rights reserved.
+ */
+package org.strongswan.android.apiclient;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.Message;
+import android.os.Messenger;
+import com.google.inject.Inject;
+
+/**
+ * @author Marcin Waligórski <marcin.waligorski@fancyfon.com>
+ */
+public class ReturnMessenger {
+    private static final String TAG = ReturnMessenger.class.getSimpleName();
+
+    @Inject
+    Context context;
+    @Inject
+    Logger logger;
+
+    public Messenger getReturnMessenger() {
+        return returnMessenger;
+    }
+
+    private Messenger returnMessenger = new Messenger(new Handler() {
+        @Override
+        public void handleMessage(Message msg) {
+            if (msg.what == getInteger(org.strongswan.android.api.R.integer.vpn_profile_create_message)) {
+                if (msg.arg2 == 0) {
+                    logger.logAndToast(TAG, "Vpn created successfully.");
+                } else {
+                    logger.logAndToast(TAG, "Vpn create failed.");
+                }
+            } else if (msg.what == getInteger(org.strongswan.android.api.R.integer.vpn_profile_read_message)) {
+            } else if (msg.what == getInteger(org.strongswan.android.api.R.integer.vpn_profile_read_all_message)) {
+                Bundle data = msg.getData();
+                long[] ids = data.getLongArray(context.getString(R.string.vpn_profile_bundle_ids_key));
+                if (ids.length == 0) {
+                    logger.logAndToast(TAG, "No vpn profiles");
+                    return;
+                }
+                for (long id : ids) {
+                    Bundle bundle = data.getBundle(context.getString(R.string.vpn_profile_bundle_id_params_key, id));
+                    logger.logAndToastVpnProfileBundle(TAG, bundle);
+                }
+            } else if (msg.what == getInteger(org.strongswan.android.api.R.integer.vpn_profile_update_message)) {
```

Comment:
Anothe empty ifs (if you are planning add here somethoing in future  maybe live comments